### PR TITLE
github action typos

### LINKED ISSUE #34: feature: GitHub action for typos
You opened a few typos fix

https://github.com/lasiar/canonicalheader/pull/26
https://github.com/lasiar/canonicalheader/pull/30

I can help by setting up either typos or codespell actions, are you interested ?

https://github.com/marketplace/actions/typos-action
https://github.com/marketplace/actions/codespell-with-annotations

### PR DESCRIPTION
**Pull request by ccoVeille**
- **Fix typos**
- **Add GitHub Typo Action**

Fixes #34

### CODE REVIEW

File: .github/workflows/spellchecker.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+jobs:
+  typos:
+    name: Spell Check with Typos
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Actions Repository
+      uses: actions/checkout@v4
+    - name: typos-action
+      uses: crate-ci/typos@v1.21.0
```
Comment (lasiar) [inline .github/workflows/spellchecker.yml:12]: Can you add newline, to the end of the file?

https://stackoverflow.com/a/729795

File: .github/workflows/spellchecker.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: Spell checking
```
Comment (lasiar) [inline .github/workflows/spellchecker.yml:1]: Other actions don't start with a capital letter

```suggestion
name: spell checking
```

**Review by lasiar — CHANGES_REQUESTED** [review]
Hi!
Great!
There are little comments.

**Comment by ccoVeille** [comment]
pushed back

**Review by lasiar — APPROVED** [review]
LGTM

**Comment by lasiar** [comment]
@ccoVeille, Thanks!

**Pull request merged**